components/feature/customtabs/src/main/java/mozilla/components/feature/customtabs/CustomTabWindowFeature.kt

```diff
@@ -57,16 +57,29 @@ class CustomTabWindowFeature(
     }
 
     /**
-     * Starts the feature and a observer to listen for window requests.
+     * Starts observing the configured session to listen for window requests.
      */
     override fun start() {
-        sessionManager.findSessionById(sessionId)?.register(windowObserver)
+        scope = store.flowScoped { flow ->
```

Comment:
So, the way I see it, the behaviour is slightly different. We won't process new state changes in the reducer before we've notified all observers. So, there's no way for a new `WindowRequest` to come in. This is the main advantage of `browser-state` over `browser-session`. We're able to shield from side-effects that observers themselves might trigger and can guarantee that things happen in a reliable order. This would be a bug in `browser-state` if it didn't work.

You're right though that we'll only process one `WindowRequest` and that was deliberate on my part and the same as we have now. That's fine for users clicking on `target="_blank" links. If a site calls `window.open` in a loop then we won't act on that, we'd only open one window. Do you consider this a problem? If so, let's file a follow-up issue, as that would be quite tricky to fix as part of the migration.

Another reason multiple `window.open` calls won't be acted on is because we immediately switch to the newly opened window/tab and since we're observing the selected session in this flow, we wouldn't be observing `window.open` of the previous tab. So, that's the way I intended for this to work.

Reply:
I'll need to think about this with a fresh brain tomorrow, but you appear confident about producer behaviour, so I'll take that :)

> Do you consider this a problem?

I'm not actually sure tbh, since I don't know how browsers normally behave in this case. But, I don't think handling "just call window.open a ton of times in a loop" is the right thing to do, so I think behaviour as-is is good.

Reply:
OK, I'll double check and test as well tomorrow before landing!